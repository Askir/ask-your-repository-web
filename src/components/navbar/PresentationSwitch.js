// @flow
/* eslint-disable jsx-a11y/no-autofocus */
import React, { Component } from 'react';
import classNames from 'classnames';
import onClickOutside from 'react-onclickoutside';
import { connect } from 'react-redux';
import { turnOnPresentationMode, turnOffPresentationMode } from '../../state/presentation_mode/presentationMode.actionCreators';
import type { AppState } from '../../state/AppState';
import type { Team } from '../../models/Team';
import './PresentationSwitch.scss';

type Props = {
  dispatch: Function,
  activeTeam: ?Team,
  isPresentationModeOn: boolean,
};

class PresentationSwitch extends Component<Props, State> {
  handleClick = () => {
    const { dispatch } = this.props;
    if (!this.props.isPresentationModeOn) {
      dispatch(turnOnPresentationMode());
    } else {
      dispatch(turnOffPresentationMode());
    }

    // this.setState(previousState => ({ isChecked: !previousState.isChecked }));
  }


  render() {
    const { isPresentationModeOn } = this.props;
    const className = classNames('PresentationSwitch', { 'PresentationSwitch--active': isPresentationModeOn });
    const outerClassName = classNames('PresentationSwitch__outer', { 'PresentationSwitch__outer--active': isPresentationModeOn });
    const innerClassName = classNames('PresentationSwitch__inner', { 'PresentationSwitch__inner--active': isPresentationModeOn });

    return (
      <button type="button" data-tip="Turn on Presentation Mode" className={className} onClick={this.handleClick}>
        <div className={outerClassName}>
          <div className={innerClassName} />
        </div>
      </button>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  activeTeam: state.activeTeam,
  isPresentationModeOn: state.presentationMode.isActive,
});

export default connect(mapStateToProps)(onClickOutside(PresentationSwitch));
