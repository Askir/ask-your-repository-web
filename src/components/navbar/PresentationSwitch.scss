.PresentationSwitch {

    margin: 0.5em;
    background-color: transparent;

    &:focus {
        outline: 0;
    }

    &__outer {
        vertical-align: middle;
        width: 47px;
        height: 24px;
        border-radius: 999px;
        background-color: #00B0F2;
        -webkit-transition-duration: .4s;
        transition-duration: .4s;
        -webkit-transition-property: background-color, box-shadow;
        transition-property: background-color, box-shadow;
        cursor: pointer;

        &--active {
            width: 47px;
            background-position: 0 0;
            background-color: #3b89ec;
        };

        &--active {
            background-color: #57BB00;
        }
    }

    &__inner {
        float: left;
        width: 23px; height: 23px;
        border-radius: inherit;
        background: #ffffff;
        -webkit-transition-timing-function: cubic-bezier(1,0,0,1);
        transition-timing-function: cubic-bezier(1,0,0,1);
        -webkit-transition-duration: 0.4s;
        transition-duration: 0.4s;
        -webkit-transition-property: transform, background-color;
        transition-property: transform, background-color;
        pointer-events: none;
        margin-top: 1px;
        margin-left: 1px;

        &--active {
            -webkit-transform: translate3d(10px, 0, 0);
            transform: translate3d(10px, 0, 0);
            background-color: #ffffff;
        };

        &--active {
            -webkit-transform: translate3d(23px, 0, 0);
            transform: translate3d(23px, 0, 0);
        }
    }
}