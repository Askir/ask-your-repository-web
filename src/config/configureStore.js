// @flow
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../state/rootReducer';

function configureStore() {
  let middleware = [thunk];

  if (process.env.NODE_ENV !== 'production') {
    const { logger } = require('redux-logger');
    middleware = [...middleware, logger];
  }

  return createStore(
    rootReducer,
    applyMiddleware(...middleware),
  );
}

export default configureStore;
