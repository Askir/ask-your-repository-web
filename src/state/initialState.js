// @flow

// This file is only for state related testing

import type { AppState } from './AppState';
import { initialState as presentation } from './presentation/presentation.reducer';
import { initialState as teamSidebar } from './team_sidebar/teamSidebar.reducer';
import { initialState as activeTeam } from './active_team/activeTeam.reducer';
import { initialState as image } from './image/image.reducer';

const initialState: AppState = {
  flash: { messages: [] },
  presentation,
  image,
  activeTeam,
  teamSidebar,
};

export default initialState;
