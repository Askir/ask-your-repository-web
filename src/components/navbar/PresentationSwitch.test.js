// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import initialState from '../../state/initialState';
import PresentationSwitch from './PresentationSwitch';

const mockStore = configureStore();

describe('<PresentationSwitch />', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow((
      <PresentationSwitch store={store} />
    )).dive();
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
