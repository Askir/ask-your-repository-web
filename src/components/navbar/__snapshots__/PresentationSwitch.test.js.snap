// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PresentationSwitch /> renders correctly 1`] = `
<button
  className="PresentationSwitch"
  data-tip="Turn on Presentation Mode"
  onClick={[Function]}
  type="button"
>
  <div
    className="PresentationSwitch__outer"
  >
    <div
      className="PresentationSwitch__inner"
    />
  </div>
</button>
`;
